osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -257,7 +257,7 @@ private void updateState(ArmedState newState, bool force = false)
                 }
             }
 
-            if (state.Value != ArmedState.Idle && LifetimeEnd == double.MaxValue || HitObject.HitWindows == null)
+            if (LifetimeEnd == double.MaxValue && (state.Value != ArmedState.Idle || HitObject.HitWindows == null))
```

Comment:
I'm pretty confident this change is more correct than what we had, but make sure to double-check you're OK with it.